### ISSUE TITLE
[Speedreader[ Crash

### ISSUE THREAD

**Issue by boocmp** [brave/brave-browser#34836]
```
[ 00 ] speedreader_ffi::speedreader::rewriter_set_min_out_length(speedreader_ffi::speedreader::CRewriter *,int) ( speedreader.rs:87 )
[ 01 ] speedreader::Rewriter::SetMinOutLength(int) ( speedreader.cc:55 )
[ 02 ] speedreader::SpeedreaderRewriterService::MakeRewriter(GURL const &,std::__Cr::basic_string<char,std::__Cr::char_traits<char>,std::__Cr::allocator<char> > const &,std::__Cr::basic_string<char,std::__Cr::char_traits<char>,std::__Cr::allocator<char> > const &,std::__Cr::basic_string<char,std::__Cr::char_traits<char>,std::__Cr::allocator<char> > const &,std::__Cr::basic_string<char,std::__Cr::char_traits<char>,std::__Cr::allocator<char> > const &) ( speedreader_rewriter_service.cc:175 )
[ 03 ] speedreader::DistillPage(GURL const &,std::__Cr::basic_string<char,std::__Cr::char_traits<char>,std::__Cr::allocator<char> >,speedreader::SpeedreaderService *,speedreader::SpeedreaderRewriterService *,base::OnceCallback<void (speedreader::DistillationResult, std::__Cr::basic_string<char,std::__Cr::char_traits<char>,std::__Cr::allocator<char> >, std::__Cr::basic_string<char,std::__Cr::char_traits<char>,std::__Cr::allocator<char> >)>) ( speedreader_util.cc:191 )
[ 04 ] std::__Cr::__tuple_impl<std::__Cr::__tuple_indices<0,1,2>,base::WeakPtr<speedreader::SpeedReaderURLLoader>,GURL,std::__Cr::basic_string<char,std::__Cr::char_traits<char>,std::__Cr::allocator<char> > >::__tuple_impl(std::__Cr::__tuple_indices<0,1,2>,std::__Cr::__tuple_types<base::WeakPtr<speedreader::SpeedReaderURLLoader>,GURL,std::__Cr::basic_string<char,std::__Cr::char_traits<char>,std::__Cr::allocator<char> > >,std::__Cr::__tuple_indices<>,std::__Cr::__tuple_types<>,base::WeakPtr<speedreader::SpeedReaderURLLoader> &&,GURL &,std::__Cr::basic_string<char,std::__Cr::char_traits<char>,std::__Cr::allocator<char> > const &) ( tuple:509 )
[ 05 ] std::__Cr::tuple<base::WeakPtr<speedreader::SpeedReaderURLLoader>,GURL,std::__Cr::basic_string<char,std::__Cr::char_traits<char>,std::__Cr::allocator<char> > >::tuple(base::WeakPtr<speedreader::SpeedReaderURLLoader> &&,GURL &,std::__Cr::basic_string<char,std::__Cr::char_traits<char>,std::__Cr::allocator<char> > const &) ( tuple:682 )
[ 06 ] speedreader::SpeedReaderURLLoader::CompleteLoading(std::__Cr::basic_string<char,std::__Cr::char_traits<char>,std::__Cr::allocator<char> >) ( speedreader_url_loader.cc:139 )
[ 07 ] body_sniffer::BodySnifferURLLoader::CheckBufferedBody(unsigned int) ( body_sniffer_url_loader.cc:174 )
[ 08 ] speedreader::SpeedReaderURLLoader::OnBodyReadable(unsigned int) ( speedreader_url_loader.cc:107 )
[ 09 ] base::internal::FunctorTraits<void (*)(const base::RepeatingCallback<void (unsigned int)> &, unsigned int, const mojo::HandleSignalsState &),void>::Invoke((base::RepeatingCallback<void (unsigned int)> const &,unsigned int,mojo::HandleSignalsState const &) const &,base::RepeatingCallback<void (unsigned int)> const &,unsigned int &&,mojo::HandleSignalsState const &) ( bind_internal.h:631 )
[ 10 ] base::internal::InvokeHelper<0,void,0>::MakeItSo((base::RepeatingCallback<void (unsigned int)> const &,unsigned int,mojo::HandleSignalsState const &) const &,std::__Cr::tuple<base::RepeatingCallback<void (unsigned int)> > const &,unsigned int &&,mojo::HandleSignalsState const &) ( bind_internal.h:868 )
[ 11 ] base::internal::Invoker<base::internal::BindState<void (*)(const base::RepeatingCallback<void (unsigned int)> &, unsigned int, const mojo::HandleSignalsState &),base::RepeatingCallback<void (unsigned int)> >,void (unsigned int, const mojo::HandleSignalsState &)>::RunImpl((base::RepeatingCallback<void (unsigned int)> const &,unsigned int,mojo::HandleSignalsState const &) const &,std::__Cr::tuple<base::RepeatingCallback<void (unsigned int)> > const &,std::__Cr::integer_sequence<unsigned long long,0>,unsigned int &&,mojo::HandleSignalsState const &) ( bind_internal.h:968 )
[ 12 ] base::internal::Invoker<base::internal::BindState<void (*)(const base::RepeatingCallback<void (unsigned int)> &, unsigned int, const mojo::HandleSignalsState &),base::RepeatingCallback<void (unsigned int)> >,void (unsigned int, const mojo::HandleSignalsState &)>::Run(base::internal::BindStateBase *,unsigned int,mojo::HandleSignalsState const &) ( bind_internal.h:932 )
[ 13 ] base::RepeatingCallback<void (unsigned int, const mojo::HandleSignalsState &)>::Run(unsigned int,mojo::HandleSignalsState const &) ( callback.h:337 )
[ 14 ] mojo::SimpleWatcher::OnHandleReady(int,unsigned int,mojo::HandleSignalsState const &) ( simple_watcher.cc:278 )
[ 15 ] base::TaskAnnotator::RunTaskImpl(base::PendingTask &) ( task_annotator.cc:201 )
```

**Comment by kjozwiak** [comment]
Moving this into https://github.com/brave/brave-browser/milestone/453 it didn't get merged into `master` and uplifted into `1.61.x` in time for the `1.61.x - Release #3`. We'll release this next week when we get the next minor `C120` bump.

**Comment by kjozwiak** [comment]
The above requires `1.61.108` or higher for `1.61.x`  verification 👍
